src/imagery/i.eodag/testsuite/test_eodag.py

```diff
@@ -0,0 +1,435 @@
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery <hamedashraf2004 gmail.com>
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Tests i.eodag input parsing / searching results.
+#              Uses NC Full data set.
+#
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+import re
+import json
+from datetime import datetime
+from grass.gunittest.case import TestCase
+from grass.gunittest.gutils import get_current_mapset, is_map_in_mapset
+from grass.exceptions import CalledModuleError
+from grass.pygrass.modules import Module
+from subprocess import PIPE
+
+
+class TestEodag(TestCase):
+
```

Comment:
```suggestion
    available_providers = {
        "peps": True,
        "cop_dataspace": True,
        "creodias": True,
        "planetary_computer": True,
        }
    search_parameters = {
        producttype="S2_MSI_L1C",
        start="2024-01-01",
        end="2024-01-01",
        }
    dag = EODataAccessGateway()
    for provider in available_providers:
        try:
            dag.set_preferred_provider(provider)
            search_result = dag.search_all(**search_parameters)
        except Exception:
            available_providers[provider] = False
```

You may have to add a case where you check if all providers are unavailable, as that indicates a situation we may have to address one way or another...